test/spec/SearchResults/getFacetValues-facetOrdering.js

```diff
@@ -0,0 +1,1518 @@
+        {count: 551, isRefined: false, name: 'Insignia™'},
+        {count: 386, isRefined: true, name: 'Apple'}
+      ]
+    ],
+    [
+      'one item ordered (sort by alpha)',
+      {
+        values: {
+          brand: {
+            order: ['Samsung'],
+            sortRemainingBy: 'alpha'
+          }
+        }
+      },
+      [
+        {count: 511, isRefined: false, name: 'Samsung'},
+        {count: 386, isRefined: true, name: 'Apple'},
+        {count: 551, isRefined: false, name: 'Insignia™'}
+      ]
+    ],
+    [
+      'one item ordered (sort by hidden)',
+      {
+        values: {
+          brand: {
+            order: ['Samsung'],
+            sortRemainingBy: 'hidden'
+          }
+        }
+      },
+      [{count: 511, isRefined: false, name: 'Samsung'}]
+    ]
+  ])('%p', function(_name, facetOrdering, expected) {
+    var data = require('./getFacetValues/disjunctive.json');
+    var order = {
+      renderingContent: {
+        facetOrdering: facetOrdering
+      }
+    };
+    var results = data.content.results.slice();
+    results[0] = Object.assign(order, results[0]);
+
+    var searchParams = new SearchParameters(data.state);
+    var result = new SearchResults(searchParams, results);
+
+    var facetValues = result.getFacetValues('brand');
+
+    expect(facetValues).toEqual(expected);
+  });
+
+  test('sortBy overrides facetOrdering', function() {
+    var data = require('./getFacetValues/disjunctive.json');
+    var order = {
+      renderingContent: {
+        facetOrdering: {
+          values: {
+            brand: {
+              order: ['Samsung', 'Apple', 'Insignia™']
+            }
+          }
+        }
+      }
+    };
+    var results = data.content.results.slice();
+    results[0] = Object.assign(order, results[0]);
+
+    var searchParams = new SearchParameters(data.state);
+    var result = new SearchResults(searchParams, results);
+
+    var facetValues = result.getFacetValues('brand', {sortBy: ['name:desc']});
+
+    var expected = [
+      {count: 511, isRefined: false, name: 'Samsung'},
+      {count: 551, isRefined: false, name: 'Insignia™'},
+      {count: 386, isRefined: true, name: 'Apple'}
+    ];
+
+    expect(facetValues).toEqual(expected);
+  });
+
+  test('facetOrdering: true overrides sortBy', function() {
+    var data = require('./getFacetValues/disjunctive.json');
+    var order = {
+      renderingContent: {
+        facetOrdering: {
+          values: {
+            brand: {
+              order: ['Samsung', 'Apple', 'Insignia™']
+            }
+          }
+        }
+      }
+    };
+    var results = data.content.results.slice();
+    results[0] = Object.assign(order, results[0]);
+
+    var searchParams = new SearchParameters(data.state);
+    var result = new SearchResults(searchParams, results);
+
+    var facetValues = result.getFacetValues('brand', {
+      sortBy: ['name:desc'],
+      facetOrdering: true
+    });
```

Comment:
That behavior is getting confusing. Shouldn't we throw instead?

Reply:
this is needed for the hierarchical menu and menu widgets, they have a sortBy which isn't the default one, but still need to take facet ordering in account.

Reply:
IIUC that's an internal constraint but that constraint leaks in the public API, which is annoying—but I get the point.

Reply:
I can see how others might want both a non-default sortBy, but still try to use facetOrdering if it's given